### ISSUE TITLE
Provide feedback for broken headings structures.

### ISSUE THREAD

**Issue by aristath** [WordPress/gutenberg#29562]
## What problem does this address?
If the user adds headings in the wrong order (for example starts with an h3 followed directly by an h6), there is no feedback about the wrong structure. 

## What is your proposed solution?
Users should be warned about skipped heading levels.
This could be done as a notification, or if they have a table-of-contents block it can happen in there as well. Since the TOC block will be parsing the headings, it will be possible to detect any abnormalities and inform the user so they can improve their structure.

**Comment by jdevalk** [comment]
Note that this will need a dev note, because this might mean some existing plugins need to update their functionality / remove their functionality that does exactly this :)

**Comment by mtias** [comment]
I think we should use the same mechanisms we have in the doc outline:

![image](https://user-images.githubusercontent.com/548849/109983364-0a882300-7d03-11eb-98b5-7148360e06c1.png)

We could fold this into this broader issue: "Notify about wrong level more clearly" #20870

**Comment by mtias** [comment]
@ZebulanStanphill yes, I think that one is mostly ready. Agreed on focusing on only block headings for now.

**Comment by paaljoachim** [comment]
How is this issue and associated PR's coming along?

**Comment by ZebulanStanphill** [comment]
@paaljoachim The UI probably needs to be redesigned again, because now the heading level options are listed vertically, rather than horizontally.

Any ideas on how it should look? I suppose you could just put the warning to the right of the options, but I'm not sure if that's the right approach. Furthermore, there was a link to a help page about headings which was always shown in the popover, and if that was kept, it would force the popover to always be much larger than it is currently.

Basically, how do I rework this old design:
![image](https://user-images.githubusercontent.com/19592990/87800330-e29b8c00-c813-11ea-838a-1a0b0240aaf3.png)

... so it works with this:
![image](https://user-images.githubusercontent.com/19592990/139141213-a2211694-9a74-411e-bc56-75cfaf643e5b.png)

**Comment by paaljoachim** [comment]
Hey @ZebulanStanphill 

I assume it would be something like this:

![Invalid-heading-levels](https://user-images.githubusercontent.com/5323259/139147804-69563287-dcd8-4a71-9a96-3f26af66ef1b.jpg)

As Joen @jasmussen has been working on the Heading block I would assume he might have some thoughts on this as well.

**Comment by skorasaurus** [comment]
@aristath : 

https://github.com/WordPress/gutenberg/issues/10581 covers this quite a bit; please reopen if there's not enough overlap between the two issues.
